docxcompose/properties.py

```diff
@@ -13,50 +68,144 @@ class CustomProperties(object):
     """
     def __init__(self, doc):
         self.doc = doc
+        self.part = None
         self._element = None
 
         try:
             part = doc.part.package.part_related_by(RT.CUSTOM_PROPERTIES)
         except KeyError:
-            pass
+            self._element = parse_xml(self._part_template())
         else:
+            self.part = part
             self._element = parse_xml(part.blob)
 
-    def dict(self):
-        """Returns a dict with all custom doc properties"""
-        if self._element is None:
-            return dict()
+    def _part_template(self):
+        template_path = os.path.join(
+            os.path.dirname(__file__), 'templates', 'custom.xml')
+        with open(template_path, 'rb') as f:
+            return f.read()
 
-        props = xpath(self._element, u'.//cp:property')
-        return {prop.get('name'): prop[0].text for prop in props}
+    def _update_part(self):
+        if self.part is None:
+            # Create a new part for custom properties
+            partname = PackURI('/docProps/custom.xml')
+            self.part = Part(
+                partname, CT.OFC_CUSTOM_PROPERTIES,
+                serialize_part_xml(self._element), self.doc.part.package)
+            self.doc.part.package.relate_to(self.part, RT.CUSTOM_PROPERTIES)
+            self._element = parse_xml(self.part.blob)
+        else:
+            self.part._blob = serialize_part_xml(self._element)
 
-    def get(self, name):
+    def __getitem__(self, key):
         """Get the value of a property."""
         prop = xpath(
             self._element,
-            u'.//cp:property[@name="{}"]'.format(name))
+            u'.//cp:property[@name="{}"]'.format(key))
+
+        if not prop:
+            raise KeyError(key)
+
+        return vt2value(prop[0][0])
+
+    def __setitem__(self, key, value):
+        """Set the value of a property."""
+        prop = xpath(
+            self._element,
+            u'.//cp:property[@name="{}"]'.format(key))
+        if not prop:
+            self.add(key, value)
+            return
+
+        el = prop[0][0]
+        new_el = value2vt(value)
+        el.getparent().replace(el, new_el)
+
+        self._update_part()
+
+    def __delitem__(self, key):
+        """Delete a property."""
+        prop = xpath(
+            self._element,
+            u'.//cp:property[@name="{}"]'.format(key))
+
+        if not prop:
+            raise KeyError(key)
+
+        prop[0].getparent().remove(prop[0])
+        # Renumber pids
+        pid = 2
```

Comment:
magic constant. why two? i'd use a static variable to describe intent here.

Reply:
No idea why. Ask Microsoft 😄 
I've moved it into a static variable.